catalogue/webapp/services/prismic/fetch/index.ts

```diff
@@ -0,0 +1,85 @@
+import * as prismic from '@prismicio/client';
+import fetch from 'node-fetch';
+import { gql, GraphQLClient } from 'graphql-request';
+import { PrismicApiError } from '../types';
+import { capitalize } from '@weco/common/utils/grammar';
+
+export const typesToPrismicGraphQLSchemaTypes = {
+  // types to graphql query schema types,
+  events: 'allEventss',
+  exhibitions: 'allExhibitionss',
+  articles: 'allArticless',
+  series: 'allSeriess',
+  webcomics: 'allWebcomicss',
+};
+
+export const prismicGraphQLQuery = (
+  type: string,
+  query?: string | string[],
+  pageSize?: number
+) => {
+  return gql`
+    query {
+      ${
+        typesToPrismicGraphQLSchemaTypes[type]
+      }(fulltext: "${query}" sortBy: title_ASC first: ${pageSize}) {
+        edges {
+          node {
+            title
+            _meta { id, firstPublicationDate }
+            contributors {
+              contributor {
+                ...on People {
+                  name
+                }
+              }
+            }
+            body {
+              ...on ${capitalize(type)}BodyStandfirst {
+                primary {
+                  text
+                }
+              }
+            }
+            promo {
+              ...on ${capitalize(type)}PromoEditorialimage {
+                primary {
+                  image
+                  link
+                  caption
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  `;
+};
+
+const endpoint = prismic.getRepositoryEndpoint('wellcomecollection');
+const client = prismic.createClient(endpoint, { fetch });
+
+export async function prismicGraphQLClient(
+  type: string,
+  query: string,
+  pageSize: number
+) {
+  const graphqlClient = new GraphQLClient(
+    prismic.getGraphQLEndpoint('wellcomecollection'),
+    {
+      method: 'get',
+      fetch: client.graphQLFetch,
+    }
+  );
+  const graphQLQuery = prismicGraphQLQuery(type, query, pageSize);
+  return graphqlClient.request(graphQLQuery);
+}
+
+export const prismicApiError = (): PrismicApiError => ({
+  errorType: 'http',
+  httpStatus: 500,
+  label: 'Internal Server Error',
+  description: '',
```

Comment:
As a non blocker sidenote; would we want a description here? This displays on the FE right, so maybe not, do we have a 500 page? 🤔

Reply:
That's a really good point, I'll make a note to have a look at what we do right now, I'm fairly sure there's something we can just use.

Reply:
Can defo be a side ticket, but something that could be useful to have!